src/lib.rs

```diff
@@ -885,7 +947,10 @@ impl<T, C: MemConfig> Extend<T> for SegVec<T, C> {
         let additional = std::cmp::max(max_size.unwrap_or(0), min_size);
         self.reserve(additional);
         for i in iter {
-            self.push(i);
+            // SAFETY: we just reserved space for these elements
+            unsafe {
+                self.push_unchecked(i);
+            }
```

Comment:
Unfortunately, I don't think we can do this.

The docs for [`size_hint`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint) say:

> size_hint() is primarily intended to be used for optimizations such as reserving space for the elements of the iterator, but must not be trusted to e.g., omit bounds checks in unsafe code. An incorrect implementation of size_hint() should not lead to memory safety violations.

So - this code can call `self.reserve(additional)` with the result of caling `size_hint` as an optimization, but we can't treat this as a guaratee that `iter` will only produce exactly `additional` elements.

Reply:
We could do one of two things:

1. Revert to the `push` implementation that was there previously, or
2. Use `push_unchecked` until we've exhausted the reserved space, then fall back to `push`

I assume the most common case will be that `size_hint` is correct and honest, and I assume that `push_count < additional` and `push_count += 1` are still faster than a checked `push`, so it seems like option 2 might be worth it. I haven't had a chance to actually benchmark this to make sure, though.

Reply:
that's true - somehow i forgot that `size_hint` is, indeed, just a hint. i think option 2 is the most reasonable, so i'll implement it and see how it goes.